### ISSUE TITLE
Implement similar widget to``AnimatedDefaultTextStyle`` but for child ``Icon``

### ISSUE THREAD

**Issue by JGeek00** [flutter/flutter#137251]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Use case

I have a ``Row``with a ``Text``and an ``Icon`` as children. I'm using ``AnimateDefaultTextStyle`` to animate the color change and the fontWeight change on the text, but I want to apply that same color animation that the text has, to the icon. 

### Proposal

``AnimateDefaultTextStyle`` should also affect to child ``Icon``. Based on the use case I commented, If I want to also animate the color change on the icon, apart of the text, I have to create an animation controller and the animation just for the icon, because the text does that with the ``AnimateDefaultTextStyle``.
The alternative would be to create a widget similar to ``AnimateDefaultTextStyle`` but for ``Icon``.

**Comment by JGeek00** [comment]
This are my two options:

**``AnimatedDefaultTextStyle`` color affects to ``Text`` and ``Icon``**
```dart
bool enabled = true;

...

AnimatedDefaultTextStyle(
  duration: ...
  style: TextStyle(
    color: enabled ? Colors.green : Colors.red
  ),
  child: Row(
    children: [
      Icon(Icons.add_rounded),
      Text("Added)"
    ]
  )
)
```

**---OR---**

**New ``AnimatedDefaultIcon`` widget to animate easily an Icon**

```dart
bool enabled = true;

...

Row(
  children: [
      AnimatedDefaultIcon(    // NEW
       duration: ...
       color: enabled ? Colors.green : Colors.red,
       size: enabled ? 24 : 20,
       child: Icon(Icons.add_rounded)
     ),
     AnimatedDefaultTextStyle(
       duration: ...
       style: TextStyle(
         color: enabled ? Colors.green : Colors.red
       ),
       child: Text("Added"),
     )
  ]
)
```

**Comment by dam-ease** [comment]
Hi @JGeek00. Thanks for your response. Looking through the use case, I think your use case can be achieved using [AnimatedIcon](https://api.flutter.dev/flutter/material/AnimatedIcon-class.html). Can you check this out please?

**Comment by JGeek00** [comment]
No, ``AnimatedIcon`` doesn't do what I want. I just want to animate the color change between 3 colors based on 3 different status (normal, active and hover). The ``AnimatedIcon`` widget requires an icon of type ``AnimatedIconData`` instead of the normal ``AnimatedIcon``, and also requires a progress.

The ``AnimatedDefaultTextStyle`` widget automatically creates the color change animation when you change the color that is passed to it. I want that same behavior for the ``Icon`` widget.

I would like to have something like this:
```dart
...

bool _isActive = false;
bool _isHover = false;

...

AnimateDefaultIcon(
  duration: Duration(seconds: 1),
  curve: Curves.ease,
  size: 20,
  color: _isHover 
      ? Colors.green
      : _isActive 
           ? Colors.red
           : Colors.blue,
  child: Icon(Icons.add)
)
```
And given this, the ``AnimateDefaultIcon`` should create an animation (with the duration and curve provided) to switch between that 3 colors based on the condition.

**Comment by HansMuller** [comment]
This is an interesting idea. I've labeled it P3 because we don't plan to pursue it in the near term. You might considering building a one-off for your app; if you're feeling ambitious we'll be happy to review a PR.

**Comment by TahaTesser** [comment]
`AnimatedDefaultTextStyle` actually updates `DefaultTextStyle` which updates the descending the text style and animates.

 However, Icons don't have a default style, it uses `IconTheme` or `Icon` widget parameters to customize itself like other widgets. If you want to animate Icons, please use `AnimatedTheme` which is effectively same as `AnimatedDefaultTextStyle`, it's suitable to animate icons when icon color, size\, and other icon properties are updated. 

I've a made a similar fix for button icons in https://github.com/flutter/flutter/pull/162667
